Rules/AvoidGlobalFunctions.cs

```diff
@@ -0,0 +1,205 @@
+﻿using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Globalization;
+using System.Management.Automation.Language;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    [Export(typeof(IScriptRule))]
+    public class AvoidGlobalFunctions : AstVisitor, IScriptRule
+    {
+        List<DiagnosticRecord> records;
```

Comment:
Please qualify the fields with the access modifiers. (by default they are private but it is good practice to be explicit)